KOGITO-5676: BPMN Editor - Containment not working when Node overlaps the Connector while splicing

handreyrc opened the pull request:
Hey @romartin, please could you review this PR?

**JIRA**: [KOGITO-5676](https://issues.redhat.com/browse/KOGITO-5676)

**VS Code**: [plugin](https://drive.google.com/file/d/1yGvvu7NSdkIwhpgRHqrcYDOdXO4VF7uV/view?usp=sharing)

Thanks

romartin reviewed (approved):
Looks good to me, thanks @handreyrc !

handreyrc commented:
Hey @romartin @LuboTerifaj,
I fixed a small side-effect, added more unit tests, rebased and updated the plugin.  This PR is ready for review and testing again.

Thanks!

LuboTerifaj commented:
Hello @handreyrc , VSCode extension is missing in both google drive you linked here and the build in PR's checks. Can you please have a look at it? Thanks!

LuboTerifaj reviewed (commented):
Hello @handreyrc , 

I couldn't have reproduce the issue on main branch. Could you please add the steps to reproduce to the jira? See also my [comment](https://issues.redhat.com/browse/KOGITO-5676?focusedCommentId=18854146&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-18854146) there.

Thank you!

handreyrc commented:
Hey @LuboTerifaj,
It's strange you could,t get the vscode plugin from my link, maybe some browser cache issue. 
Anyway here goes a new link for the plugin:

**VS Code**: [plugin](https://drive.google.com/file/d/1yGvvu7NSdkIwhpgRHqrcYDOdXO4VF7uV/view?usp=sharing) 

I already updated the issue with the steps to reproduce it as well, please ping me if you can't reproduce it yet.

The FDB is failing due to some unrelated issue, I'll keep an eye to get it green.

Thanks for checking it!

karreiro reviewed (approved):
Thank you, @handreyrc! Great stuff :-)

handreyrc commented:
Hey @LuboTerifaj we have all green here.
I tried to answer your questions in the JIRA the best I could I hope it helps.
I tested the lastest main and I could reproduce the issue. 
Please ping me if you can't reproduce it yet and we can do a quick call.

Thanks!

LuboTerifaj commented on lienzo-tests/src/test/java/com/ait/lienzo/client/core/shape/wires/handlers/impl/WiresLineSpliceControlImplTest.java:
```diff
@@ -645,6 +647,66 @@ public void testExecuteSpliceWithContainment() {
+    }
+
+    @Test
+    public void testExecuteSpliceWithContainmentWithDeviation() {
```
There are many common lines of code in `testExecuteSpliceWithContainmentWithoutDeviation` and `testExecuteSpliceWithContainmentWithDeviation` test methods. Would it make sense to you to refactor it a little bit? You could create a method for common code for example.

LuboTerifaj reviewed (changes requested):
Hello @handreyrc , I have updated the steps to reproduce in jira.

I didn't find any issues. The fix works as expected.

However, I have one comment regarding code.

I have found related issue regarding the sequence flow colour inside containers (present also on main branch), but I have reported it in separate jira:

- [KOGITO-5778](https://issues.redhat.com/browse/KOGITO-5778) Sequence flow is red during Line splicing inside containers

Thank you!

LuboTerifaj reviewed (approved):
Hello @handreyrc , The extension you have linked in the comment above works as expected. Thanks!

romartin commented:
Hey @handreyrc , Can I merge this? I see the FDB failing, is it something related to this PR or we can go ahead? Thx

handreyrc commented:
Hey @romartin,

This PR is rebased and finally has a green FDB. It is ready to merge!

Thanks!
